### PR TITLE
Remove public db record

### PR DESCRIPTION
**Pull request by davisp**
COUCHDB-3288

### CODE REVIEW

**Review by iilyak — APPROVED** [review]
+1

File: src/cassim_security.erl [hunk 1]
```diff
@@ -90,28 +90,30 @@ set_security(DbName, SecProps) ->
...
 
-set_security(#db{name=DbName0}, #doc{}=SecDoc0, Options) ->
+set_security(DbName0, #doc{}=SecDoc0, Options) when is_binary(DbName0) ->
     DbName = mem3:dbname(DbName0),
     MetaId = cassim_metadata_cache:security_meta_id(DbName),
     SecDoc = SecDoc0#doc{id=MetaId},
     UserCtx = couch_util:get_value(user_ctx, Options, #user_ctx{}),
     MetaDbName = cassim_metadata_cache:metadata_db(),
-    MetaDb = #db{name=MetaDbName, user_ctx=?ADMIN_USER},
+    MetaDb = couch_db:clustered_db(MetaDbName, ?ADMIN_USER),
```
Comment (nickva) [inline src/cassim_security.erl]: `couch_db:clustered_db/2` returns `{ok, Db}` ?

File: src/cassim_security.erl [hunk 2]
```diff
@@ -90,28 +90,30 @@ set_security(DbName, SecProps) ->
...
+    set_security(couch_db:name(Db), SecDoc, Options).
 
 
 migrate_security_props(DbName0, {SecProps}) ->
     DbName = mem3:dbname(DbName0),
     MetaId = cassim_metadata_cache:security_meta_id(DbName),
     SecDoc = #doc{id=MetaId, body={SecProps}},
     MetaDbName = cassim_metadata_cache:metadata_db(),
-    MetaDb = #db{name=MetaDbName, user_ctx=?ADMIN_USER},
+    MetaDb = couch_db:clustered_db(MetaDbName, ?ADMIN_USER),
```
Comment (nickva) [inline src/cassim_security.erl]: `{ok, MetaDb} = ...`

**Review by nickva — CHANGES_REQUESTED** [review]
Need to fix clustered_db call.

Created a _metadata db and then did a GET on _security. Migration seems to have run and crashed.

```
[error] 2017-09-20T15:10:56.149271Z node1@127.0.0.1 <0.3791.0> 13024e8a63 req_err(4192660266) unknown_error : function_clause
    [<<"chttpd_db:update_doc/4 L1004">>,<<"cassim_security:migrate_security_props/2 L118">>,<<"cassim_security:get_security_doc/2 L72">>,<<"cassim_security:get_security/2 L45">>,<<"cloudant_auth:db_authorization_check/1 L82">>,<<"cloudant_auth:authorize_request/1 L18">>,<<"cloudant_plugin_chttpd_auth:authorize/1 L15">>,<<"couch_epi_functions_gen:apply_while/4 L278">>]
[info] 2017-09-20T15:10:56.149430Z node1@127.0.0.1 <0.3791.0> 13024e8a63 13024e8a63 127.0.0.1  localhost:15984 GET /db/_security 500 ok 2
```

**Comment by davisp** [comment]
@nickva Fixed ty

**Comment by nickva** [comment]
Now works. +1

**Pull request merged**